### ISSUE TITLE
Rawhide ostree.hotfix failure due to selinux-policy upgrade

### ISSUE THREAD

**Issue by prestist** [coreos/fedora-coreos-tracker#1364]
### Describe the bug

When building the latest version of rawhide since 12/08/22 we get a timeout on the 'ostree.hofix' kola test.

This prevents the latest rawhide build to be completed. 

### Reproduction steps

1.  cosa init https://github.com/coreos/fedora-coreos-config --branch rawhide
2. cosa fetch
3. cosa build
4. cosa kola run ostree.hotfix

### Expected behavior

The test should pass.

### Actual behavior

Test times out and fails with the following logs

```
=== RUN   ostree.hotfix
08:11:45  === RUN   ostree.hotfix/unlock
08:11:48  === RUN   ostree.hotfix/install
08:11:48  === RUN   ostree.hotfix/uninstall
08:11:49  === RUN   ostree.hotfix/persist
08:23:25  --- FAIL: ostree.hotfix (720.46s)
08:23:25      --- PASS: ostree.hotfix/unlock (5.01s)
08:23:25      --- PASS: ostree.hotfix/install (0.52s)
08:23:25      --- PASS: ostree.hotfix/uninstall (0.33s)
08:23:25      --- FAIL: ostree.hotfix/persist (612.68s)
08:23:25              unlock.go:240: Failed to reboot machine: machine "95f91dad-1372-4075-99ae-e824f3a39bc2" failed to start: ssh journalctl failed: time limit exceeded
08:23:25          cluster.go:184: "logger --tag kola '=== DONE: ostree.hotfix/persist ==='" failed: output , status ssh: handshake failed: read tcp 127.0.0.1:56220->127.0.0.1:43907: read: connection reset by peer
08:23:25          harness.go:1501: Found emergency shell on machine 95f91dad-1372-4075-99ae-e824f3a39bc2 console
08:23:25          harness.go:1501: Found systemd generator failure (/usr/lib/systemd/system-generators/coreos-installer-generator) on machine 95f91dad-1372-4075-99ae-e824f3a39bc2 console
08:23:25  FAIL, output in /home/jenkins/agent/workspace/build/tmp/kola-lW8u1/kola/rerun
08:23:25  Error: harness: test suite failed
08:23:25  2022-12-08T13:23:19Z cli: harness: test suite failed
08:23:25  error: failed to execute cmd-kola: exit status 1
```

### System details

arch= x86_64

### Additional information

After debugging, I was able to determane if rolling back the `selinux-policy` to `selinux-policy-38.1-1.fc38` I was able to get a passing build.

```
pushd overrides/rpm
koji download-build selinux-policy-38.1-1.fc38
popd

```
[console.txt](https://github.com/coreos/fedora-coreos-tracker/files/10248055/console.txt)
[journal.txt](https://github.com/coreos/fedora-coreos-tracker/files/10248053/journal.txt)

**Comment by dustymabe** [comment]
Pin in https://github.com/coreos/fedora-coreos-config/pull/2135

Next up is opening a BZ against SELinux in Fedora.

**Comment by dustymabe** [comment]
Good news: We finally got a successful `rawhide` build! Bad news: the next round of package updates included packages that need a newer selinux-policy than the one we pinned.

I'll try to pin those too, to get rawhide unblocked again.

**Comment by jlebon** [comment]
Looks like everything goes south after rebooting into a hotfix deployment. After pivoting, systemd can't even execute the generators:

```
[    3.518065] systemd[567]: Failed to execute /usr/lib/systemd/system-generators/coreos-boot-mount-generator: Permission denied
[    3.523768] systemd[568]: Failed to execute /usr/lib/systemd/system-generators/coreos-installer-generator: Permission denied
[    3.530979] systemd[570]: Failed to execute /usr/lib/systemd/system-generators/coreos-platform-chrony: Permission denied
[    3.535400] systemd[569]: Failed to execute /usr/lib/systemd/system-generators/coreos-liveiso-autologin-generator: Permission denied
[    3.542057] systemd[571]: Failed to execute /usr/lib/systemd/system-generators/kdump-dep-generator.sh: Permission denied
[    3.546922] systemd[574]: Failed to execute /usr/lib/systemd/system-generators/selinux-autorelabel-generator.sh: Permission denied
```

So this is possibly something to do with overlay + SELinux.

**Comment by prestist** [comment]
Thank you @jlebon I will be sure to add a note about that in the BZ!

**Comment by dustymabe** [comment]
> Since we have `selinux-policy` pinned in `rawhide` because of this issue we don't have the fix for #1359 in `rawhide` so we'll need to disable the `ext.config.ntp.timesyncd.dhcp-propagation` test in `rawhide` until this is resolved.

denylist entry addition in https://github.com/coreos/fedora-coreos-config/pull/2154

**Comment by dustymabe** [comment]
This was fixed upstream in https://github.com/fedora-selinux/selinux-policy/pull/1574 and landed in [`selinux-policy-38.6-1.fc38 `](https://bodhi.fedoraproject.org/updates/FEDORA-2023-849e0c6945).

**Comment by dustymabe** [comment]
Dropping the pin in https://github.com/coreos/fedora-coreos-config/pull/2209
